### PR TITLE
feat: add performance tables

### PR DESCRIPTION
**Pull request by MrtinoRG**
## Summary by Sourcery

Add scripts to generate and collect performance tables for various models, including human-scored questions, and produce LaTeX tables for performance metrics.

New Features:
- Introduce scripts to generate performance tables from model scores, including human-scored questions.

Enhancements:
- Add functionality to collect and process performance scores for various models, including human baseline comparisons.

Documentation:
- Generate LaTeX tables for performance metrics, including calculation, knowledge, reasoning, intuition, and difficulty levels.

### CODE REVIEW

**Comment by MrtinoRG** [comment]
{735303D4-4E9D-4DDF-AC18-2019D7435950}

This is the performance table for the entire dataset

{BC13D05E-0427-49A0-AC4D-55F3327D6AC0}

And this is the one for the human subset.

Tell me if you think something should be changed

**Comment by AdrianM0** [comment]
I like it!

**Review by AdrianM0 — APPROVED** [review]

File: src/scripts/collect_performance_table_scores.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    chembench_repo = obtain_chembench_repo()
+    reports = glob(
+        os.path.join(chembench_repo, "reports",
+                     "**", "reports", "**", "*.json")
+    )
+    overall_files = glob(
+        os.path.join(chembench_repo, "reports", "**", "*.json")
+    )
+
+    models = list(set([Path(p).parent for p in reports]))
```
Comment (kjappelbaum) [inline src/scripts/collect_performance_table_scores.py:198]: cool! in principle you can also always rely on `data/ model_score_dicts.pkl` 

here is an example https://www.dropbox.com/scl/fi/aszdlklymc6q0ec3z7ftz/model_score_dicts.pkl?rlkey=p4lv9gccqx8bhejgp3jaaspgp&dl=0

File: src/scripts/collect_performance_table_scores.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            overall_scores.append({
+                "model": p,
+                "scores": model_results,
+            })
+            overall_human_scores.append({
+                "model": p,
+                "scores": human_results,
+            })
+        except Exception as e:
+            logger.error(f"Error processing {file}: {e}")
```
Comment (kjappelbaum) [inline src/scripts/collect_performance_table_scores.py:235]: here you'd have 

```
src/data/humans_as_models_scores_combined.pkl
src/data/humans_as_models_scores_no_tools.pkl
src/data/humans_as_models_scores_tools.pkl
```

File: src/scripts/collect_performance_table_scores.py (hunk 1 above)
Comment (kjappelbaum) [reply src/scripts/collect_performance_table_scores.py:198]: but I can also adjust this later

File: src/scripts/make_performance_tables.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+            r"\cmidrule(lr){2-4} \cmidrule(lr){5-7}\\"
+        )
+        latex_table = latex_table.replace(
+            r"\begin{tabular}{llllllll}",
+            r"\begin{tabular}{cccccccc}"
+        )
+
+    # Wrap the table with \resizebox
+    latex_table = (
+        r"\begin{table}[htbp]\centering" + "\n" +
```
Comment (kjappelbaum) [inline src/scripts/make_performance_tables.py:275]: I think I'd only put the `latex_table` in the file and then handle caption & co in the latex file

File: src/scripts/make_performance_tables.py [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+    
+    # Round all numeric columns to 2 decimal places (excluding non-numeric columns like "Model")
+    numeric_columns = df.select_dtypes(include='number').columns
+    df[numeric_columns] = df[numeric_columns].applymap(lambda x: f"{x:.2f}".rstrip('0').rstrip('.'))
+
+
+   # Function to bold the best value (assuming higher is better)
+    def bold_best(series):
+        best = series.astype(float).max()  # Modify as per your criteria (e.g., use min() for smaller is better)
+        return series.apply(lambda x: f"\\textbf{{{x}}}" if float(x) == best else x)
```
Comment (kjappelbaum) [inline src/scripts/make_performance_tables.py:220]: this is very cool! super nice that this bolding happens automatically!

**Comment by MrtinoRG** [comment]
I have now removed all the additional processing of the `latex_table` remaining only the table itself (no caption or re-sizing).

Also, I added sorting so the table will be sorted alphabetically. I did not change how to collect the results because that little adjustments can be done afterwards

**Comment by kjappelbaum** [comment]
thanks! I'll merge now and try to massage it into a good place

**Comment by kjappelbaum** [comment]
I'm adding them into `Snakemake` at the moment. Then they will be always rerun if there is a change

**Pull request merged**